File: back/src/main/java/com/ssafy/record/domain/feed/entity/Feed.java
//package com.ssafy.record.domain.feed.entity;
//
//import com.ssafy.record.domain.user.entity.User;
//import jakarta.persistence.*;
//import lombok.*;
//
//import java.time.LocalDateTime;
//import java.util.ArrayList;
//import java.util.List;
//
//@Entity
//@Getter
//@NoArgsConstructor(access = AccessLevel.PROTECTED)
//@AllArgsConstructor
//@Builder
//public class Feed {
//
//    @Id
//    @GeneratedValue(strategy = GenerationType.IDENTITY)
//    private Long noteId;
//
//    private String content;
//
//    private Integer successCodeStart;
//    private Integer successCodeEnd;
//
//    private Integer failCodeStart;
//    private Integer failCodeEnd;
//
//    private Boolean isPublic;
//
//    private LocalDateTime createdAt;
//    private LocalDateTime updatedAt;
//
//    private Integer viewCount;
//
//    private Integer likeCount;
//
//    private Integer commentCount;
//
//    @ManyToOne(fetch = FetchType.LAZY)
//    @JoinColumn(name = "user_id")
//    private User user;
//
//    private Long problemId;
//    private String problemName;
//    private Integer problemTier;
//
//    @ElementCollection
//    private List<String> tags = new ArrayList<>();
//
//    @OneToMany(mappedBy = "feed", cascade = CascadeType.ALL, orphanRemoval = true)
//    private List<Like> likes = new ArrayList<>();
//
//    @OneToMany(mappedBy = "feed", cascade = CascadeType.ALL, orphanRemoval = true)
//    private List<Comment> comments = new ArrayList<>();
//
//    public void increaseLikeCount() {
//        this.likeCount = (this.likeCount == null ? 1 : this.likeCount + 1);
//    }
//
//    public void decreaseLikeCount() {
//        this.likeCount = (this.likeCount == null || this.likeCount <= 0) ? 0 : this.likeCount - 1;
//    }
//
//    public void increaseCommentCount() {
//        this.commentCount = (this.commentCount == null ? 1 : this.commentCount + 1);
//    }
//
//    public void decreaseCommentCount() {
//        this.commentCount = (this.commentCount == null || this.commentCount <= 0) ? 0 : this.commentCount - 1;
//    }
//}

File: front/src/types/feed.ts
/**
 * 백엔드에서 받아오는 API 피드 데이터 원형
 */
export interface ApiFeed {
  noteId: number;
  noteTitle: string;
  content: string;
  successCode: string;
  successCodeStart: number;
  successCodeEnd: number;
  successLanguage: string;
  failCode: string;
  failCodeStart: number;
  failCodeEnd: number;
  failLanguage: string;
  isPublic: boolean,
  createdAt: string;
  updatedAt: string;
  viewCount: number;
  likeCount: number;
  commentCount: number;
  user: {
    userId: number;
    bojId: string;
    nickname: string;
    userTier: number;
    image?: string;
  };
  problem: {
    problemId: number;
    problemName: string;
    problemTier: number;
  };
  tags: {
    tagId: number;
    tagName: string;
  }[];
  deleted: boolean,
  liked: boolean;
  following: boolean;
}

/**
 * 메인 피드 페이지에서 사용하는 전체 데이터 구조
 */
export interface MainFeedData {
  noteId: number;
  noteTitle: string;
  content: string;
  successCode: string;
  successCodeStart: number;
  successCodeEnd: number;
  successLanguage: string;
  failCode: string;
  failCodeStart: number;
  failCodeEnd: number;
  failLanguage: string;
  isPublic: boolean,
  createdAt: string;
  updatedAt: string;
  viewCount: number;
  likeCount: number;
  commentCount: number;
  user: {
    userId: number;
    bojId: string;
    nickname: string;
    userTier: number;
    image?: string;
  };
  problem: {
    problemId: number;
    problemName: string;
    problemTier: number;
  };
  tags: string[];
  deleted: boolean;
  liked: boolean;
  following: boolean;
}

/**
 * 탐색 페이지 피드 카드 구조
 */
export interface ExploreFeedCardData {
  noteId: number;
  noteTitle: string;
  successLanguage: string;
  isPublic: boolean,
  createdAt: string;
  viewCount: number;
  likeCount: number;
  commentCount: number;
  user: {
    userId: number;
    bojId: string;
    nickname: string;
    userTier: number;
    image?: string;
  };
  problem: {
    problemId: number;
    problemName: string;
    problemTier: number;
  };
  tags: string[];
  deleted: boolean;
  liked: boolean;
  following: boolean;
}

/**
 * 피드 정렬 옵션
 */
export type SortOption = 'latest' | 'views' | 'likes' | 'comments';
